### PR TITLE
specify the supported landscaper version as a component reference

### PR DESCRIPTION
**Pull request by reshnm**
**What this PR does / why we need it**:

Whit this PR the user has no longer to specify the version and component name of the Landscaper that shall be deployed.
The supported version of the Landscaper is now specified as a component reference of the landscaper service controller.
The landscaper service controller will automatically reconcile all deployments to match their deployed landscaper version with the one spcified in the component reference.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
The supported version of the Landscaper is now specified as a component reference of the landscaper service controller.
The landscaper service controller will automatically reconcile all deployments to match their deployed landscaper version with the one spcified in the component reference.
```

### CODE REVIEW

File: .landscaper/blueprint/deploy-execution.yaml [hunk 1]
```diff
@@ -43,6 +51,14 @@ deployItems:
...
             forceUpdate: true
 
+          landscaperServiceComponent:
+            name: {{ $landscaperServiceCompinentName }}
+            version: {{ $landscaperServiceCompinentVersion }}
+
+            repositoryContext:
+              type: {{ $repositoryContextType }}
+              baseUrl: {{ $repositoryContextBaseUrl }}
+
```
[the remark this answers is not in the document]
Comment (reshnm) [reply .landscaper/blueprint/deploy-execution.yaml:61]: Yes, this is the GCR repository. No credentials are needed to access it, it has anonymous access.

File: docs/gettingstarted/install-landscaper-service.md [hunk 2]
```diff
@@ -11,23 +17,8 @@ Please consult the [landscaper documentation](https://github.com/gardener/landsc
...
 
-The landscaper service can be installed via [Helm](https://helm.sh) using the Helm chart [charts/landscaper-service](../../charts/landscaper-service)
-or via a landscaper installation using the _landscaper-service-blueprint_ blueprint of the _github.com/gardener/landscaper-service_ component.
-
-
-### Helm
-
-To install the landscaper service via Helm, the chart has to be pulled from the landscaper service OCI registry.
-
-:warning: Attention: Helm version v3.7.0 or later is required for this to work.
```
[the remark this answers is not in the document]
Comment (reshnm) [reply docs/gettingstarted/install-landscaper-service.md:22]: The helm chart now needs the component name and the component version as well as the repository context configured in its value.
The values are configured in the landscaper service controller CD.
When installing via an installation/CD these values are automatically put into the helm chart during installation.
When doing a "manual" helm install, the user has to figure out these values on his own, which could lead to issues (landscaper version not compatible etc.).
Therefore I would prefer to have just one supported way of installing the controller.

File: .landscaper/blueprint/deploy-execution.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (achimweigel) [reply .landscaper/blueprint/deploy-execution.yaml:61]: But if we want to run LaaS for SAP, we need to move all resources into internal repos. They have to be scanned, patched, signed etc. The internal repos do not have anonymous access.

File: docs/gettingstarted/install-landscaper-service.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (achimweigel) [reply docs/gettingstarted/install-landscaper-service.md:22]: Ok :-)

File: .landscaper/blueprint/deploy-execution.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (reshnm) [reply .landscaper/blueprint/deploy-execution.yaml:61]: Ok, makes sense. Then I will add a configuration option for secrets.

**Review by achimweigel — APPROVED** [review]
/lgtm

**Pull request merged**